velox/expression/FunctionSignature.h

```diff
@@ -230,14 +208,22 @@ TypeSignature parseTypeSignature(const std::string& signature);
 class FunctionSignatureBuilder {
  public:
   FunctionSignatureBuilder& typeVariable(std::string name) {
-    typeVariableConstants_.emplace_back(name);
+    typeVariableConstraints_.emplace_back(
+        name, "", ParameterType::TYPE_PARAMETER);
+    return *this;
+  }
+
+  FunctionSignatureBuilder& integerVariable(std::string name) {
+    typeVariableConstraints_.emplace_back(
+        name, "", ParameterType::INTEGER_PARAMETER);
     return *this;
   }
 
-  FunctionSignatureBuilder& variableConstraint(
+  FunctionSignatureBuilder& integerVariableWithConstraint(
```

Comment:
Can we just add an std::optional<std::string> parameter defaulted to std::nullopt to integerVariable?